[BUG] Unable to install matrix-puppet-discord

### Describe the bug

a script error occurs during installation

### Context

- Hardware: Proxmox with Debian 11
- YunoHost version: 11.0.5
- I have access to my server: SSH / Web
- Are you in a special context or did you perform some particular tweaking on your YunoHost instance?: NO

### Steps to reproduce

only Application install
Search Matrix
Start Install Matrix Discord
Error

### Logs


============
args:
  app: matrix-puppet-discord
  force: false
  label: Matrix-Discord bridge
  no_remove_on_failure: false
ended_at: 2022-03-18 07:11:37.067934
env:
  YNH_APP_ARG_SYNAPSENUMBER: '1'
  YNH_APP_BASEDIR: /var/cache/yunohost/app_tmp_work_dirs/app_z4ongaiv
  YNH_APP_ID: matrix-puppet-discord
  YNH_APP_INSTANCE_NAME: matrix-puppet-discord
  YNH_APP_INSTANCE_NUMBER: '1'
  YNH_APP_MANIFEST_VERSION: 0.1.6~ynh2
  YNH_ARCH: amd64
error: An error occurred inside the app installation script
interface: api
operation: app_install
parent: null
related_to:
- - app
  - matrix-puppet-discord
started_at: 2022-03-18 07:11:17.619249
success: false
yunohost_version: 11.0.5

============

2022-03-18 07:11:35,240: DEBUG - npm WARN mx-puppet-discord@0.0.0 No repository field.
2022-03-18 07:11:35,249: DEBUG - npm WARN mx-puppet-discord@0.0.0 No license field.
2022-03-18 07:11:35,249: DEBUG - 
2022-03-18 07:11:35,263: DEBUG - npm ERR! Error while executing:
2022-03-18 07:11:35,263: DEBUG - npm ERR! /usr/bin/git ls-remote -h -t git://github.com/Sorunome/discord-markdown.git
2022-03-18 07:11:35,263: DEBUG - npm ERR!
2022-03-18 07:11:35,263: DEBUG - npm ERR! fatal: remote error:
2022-03-18 07:11:35,263: DEBUG - npm ERR!   The unauthenticated git protocol on port 9418 is no longer supported.
2022-03-18 07:11:35,263: DEBUG - npm ERR! Please see https://github.blog/2021-09-01-improving-git-protocol-security-github/ for more information.
2022-03-18 07:11:35,263: DEBUG - npm ERR!
2022-03-18 07:11:35,263: DEBUG - npm ERR! exited with error code: 128
2022-03-18 07:11:35,552: DEBUG - 
2022-03-18 07:11:35,552: DEBUG - npm ERR! A complete log of this run can be found in:
2022-03-18 07:11:35,552: DEBUG - npm ERR!     /opt/yunohost/matrix-puppet-discord/.npm/_logs/2022-03-18T07_11_35_271Z-debug.log
2022-03-18 07:11:35,563: DEBUG - + ynh_exit_properly
2022-03-18 07:11:37,067: ERROR - Unable to install matrix-puppet-discord: An error occurred inside the app installation script

[epawexuqef.txt](https://github.com/YunoHost-Apps/matrix-puppet-discord_ynh/files/8302437/epawexuqef.txt)

--- a/README.md
+++ b/README.md
@@ -15,7 +15,7 @@ If you don't have YunoHost, please consult [the guide](https://yunohost.org/#/in
 
 ## Overview
 
-This is a discord puppeting bridge for matrix. It handles bridging private and group DMs, as well as Guilds (servers). It is based on [mx-puppet-bridge](https://github.com/Sorunome/mx-puppet-bridge).
+This is a discord puppeting bridge for matrix. It handles bridging private and group DMs, as well as Guilds (servers). It is based on [mx-puppet-bridge](https://gitlab.com/mx-puppet/mx-puppet-bridge).
 
 ### Features
 
@@ -37,7 +37,7 @@ This is a discord puppeting bridge for matrix. It handles bridging private and g
 
 ### How to
 
-See <https://github.com/matrix-discord/mx-puppet-discord> for documentation.
+See <https://gitlab.com/mx-puppet/discord/mx-puppet-discord> for documentation.
 
 
 **Shipped version:** 0.1.7~ynh1

--- a/README.md
+++ b/README.md
@@ -40,7 +40,7 @@ This is a discord puppeting bridge for matrix. It handles bridging private and g
 See <https://github.com/matrix-discord/mx-puppet-discord> for documentation.
 
 
-**Shipped version:** 0.1.6~ynh2
+**Shipped version:** 0.1.7~ynh1
 
 ## Screenshots
 
@@ -54,9 +54,9 @@ You also may modify the `/etc/matrix-puppet-discord/user.config.yaml` file to cu
 
 ## Documentation and resources
 
-* Official app website: <https://github.com/matrix-discord/mx-puppet-discord>
-* Official admin documentation: <https://github.com/matrix-discord/mx-puppet-discord>
-* Upstream app code repository: <https://github.com/matrix-discord/mx-puppet-discord>
+* Official app website: <https://gitlab.com/mx-puppet/discord/mx-puppet-discord>
+* Official admin documentation: <https://gitlab.com/mx-puppet/discord/mx-puppet-discord>
+* Upstream app code repository: <https://gitlab.com/mx-puppet/discord/mx-puppet-discord>
 * YunoHost documentation for this app: <https://yunohost.org/app_matrix-puppet-discord>
 * Report a bug: <https://github.com/YunoHost-Apps/matrix-puppet-discord_ynh/issues>
 

--- a/README_fr.md
+++ b/README_fr.md
@@ -40,7 +40,7 @@ Ceci est une passerelle vers Discord pour Matrix. Il gère la messagerie privée
 Voir la documentation sur <https://github.com/matrix-discord/mx-puppet-discord> (en anglais).
 
 
-**Version incluse :** 0.1.6~ynh2
+**Version incluse :** 0.1.7~ynh1
 
 ## Captures d'écran
 
@@ -54,9 +54,9 @@ Vous pouvez par ailleurs modifier le fichier `/etc/matrix-puppet-discord/user.co
 
 ## Documentations et ressources
 
-* Site officiel de l'app : <https://github.com/matrix-discord/mx-puppet-discord>
-* Documentation officielle de l'admin : <https://github.com/matrix-discord/mx-puppet-discord>
-* Dépôt de code officiel de l'app : <https://github.com/matrix-discord/mx-puppet-discord>
+* Site officiel de l'app : <https://gitlab.com/mx-puppet/discord/mx-puppet-discord>
+* Documentation officielle de l'admin : <https://gitlab.com/mx-puppet/discord/mx-puppet-discord>
+* Dépôt de code officiel de l'app : <https://gitlab.com/mx-puppet/discord/mx-puppet-discord>
 * Documentation YunoHost pour cette app : <https://yunohost.org/app_matrix-puppet-discord>
 * Signaler un bug : <https://github.com/YunoHost-Apps/matrix-puppet-discord_ynh/issues>
 

--- a/README_fr.md
+++ b/README_fr.md
@@ -15,7 +15,7 @@ Si vous n'avez pas YunoHost, regardez [ici](https://yunohost.org/#/install) pour
 
 ## Vue d'ensemble
 
-Ceci est une passerelle vers Discord pour Matrix. Il gère la messagerie privée ou en groupe, tout comme la discussion dans une guilde (serveur Discord). Le projet est basé sur [mx-puppet-bridge](https://github.com/Sorunome/mx-puppet-bridge).
+Ceci est une passerelle vers Discord pour Matrix. Il gère la messagerie privée ou en groupe, tout comme la discussion dans une guilde (serveur Discord). Le projet est basé sur [mx-puppet-bridge](https://gitlab.com/mx-puppet/mx-puppet-bridge).
 
 ### Fonctionnalités
 
@@ -37,7 +37,7 @@ Ceci est une passerelle vers Discord pour Matrix. Il gère la messagerie privée
 
 ### Comment faire
 
-Voir la documentation sur <https://github.com/matrix-discord/mx-puppet-discord> (en anglais).
+Voir la documentation sur <https://gitlab.com/mx-puppet/discord/mx-puppet-discord> (en anglais).
 
 
 **Version incluse :** 0.1.7~ynh1

--- a/check_process
+++ b/check_process
@@ -20,13 +20,9 @@
 		setup_private=0
 		setup_public=0
 		upgrade=1
-		upgrade=1	from_commit=1cdeb9f003f07c4df897f29dd4a521c595327415
 		backup_restore=1
 		multi_instance=1
 		port_already_use=1
 		change_url=0
 		actions=0
 		config_panel=0
-;;; Upgrade options
-	; commit=1cdeb9f003f07c4df897f29dd4a521c595327415
-		name=Documentation

--- a/conf/app.src
+++ b/conf/app.src
@@ -1,4 +1,4 @@
-SOURCE_URL=https://gitlab.com/mx-puppet/discord/mx-puppet-discord/-/archive/master/mx-puppet-discord-master.tar.gz
+SOURCE_URL=https://gitlab.com/mx-puppet/discord/mx-puppet-discord/-/archive/3c82530ac4978ae206f63634a319523e5bff299a/mx-puppet-discord-master.tar.gz
 SOURCE_SUM=3e774c0a7fe02ba92fb2b2e45edb5601e85dfe1a164cba20b60050bfdcb9c77f
 SOURCE_SUM_PRG=sha256sum
 SOURCE_FORMAT=tar.gz

--- a/conf/app.src
+++ b/conf/app.src
@@ -1,5 +1,5 @@
-SOURCE_URL=https://gitlab.com/mx-puppet/discord/mx-puppet-discord/-/archive/fd4402268e9a8b942ca632f3337702862e06d24f/mx-puppet-discord-fd4402268e9a8b942ca632f3337702862e06d24f.tar.gz
-SOURCE_SUM=b0edc4dfe9d3670eb8b53dc229a8fd7fb794ada5fe1f3d3ba4c38b54ce5ca176
+SOURCE_URL=https://gitlab.com/mx-puppet/discord/mx-puppet-discord/-/archive/3c82530ac4978ae206f63634a319523e5bff299a/mx-puppet-discord-master.tar.gz
+SOURCE_SUM=3e774c0a7fe02ba92fb2b2e45edb5601e85dfe1a164cba20b60050bfdcb9c77f
 SOURCE_SUM_PRG=sha256sum
 SOURCE_FORMAT=tar.gz
 SOURCE_IN_SUBDIR=true

--- a/conf/app.src
+++ b/conf/app.src
@@ -1,5 +1,5 @@
 SOURCE_URL=https://gitlab.com/mx-puppet/discord/mx-puppet-discord/-/archive/3c82530ac4978ae206f63634a319523e5bff299a/mx-puppet-discord-master.tar.gz
-SOURCE_SUM=3e774c0a7fe02ba92fb2b2e45edb5601e85dfe1a164cba20b60050bfdcb9c77f
+SOURCE_SUM=df351ace3788ad9c2d932bcdfc0ba691cd071575d7bd0f9180fbe6976dca5c42
 SOURCE_SUM_PRG=sha256sum
 SOURCE_FORMAT=tar.gz
 SOURCE_IN_SUBDIR=true

--- a/conf/app.src
+++ b/conf/app.src
@@ -1,5 +1,5 @@
-SOURCE_URL=https://github.com/matrix-discord/mx-puppet-discord/archive/bb6438a504182a7a64048b992179427587ccfded.tar.gz
-SOURCE_SUM=850e2ed4ca236bdae3409f8ccb59b3ebbf00e07be65990f8b00019e9732ced26
+SOURCE_URL=https://gitlab.com/mx-puppet/discord/mx-puppet-discord/-/archive/master/mx-puppet-discord-master.tar.gz
+SOURCE_SUM=3e774c0a7fe02ba92fb2b2e45edb5601e85dfe1a164cba20b60050bfdcb9c77f
 SOURCE_SUM_PRG=sha256sum
 SOURCE_FORMAT=tar.gz
 SOURCE_IN_SUBDIR=true

--- a/conf/app.src
+++ b/conf/app.src
@@ -1,5 +1,5 @@
-SOURCE_URL=https://gitlab.com/mx-puppet/discord/mx-puppet-discord/-/archive/3c82530ac4978ae206f63634a319523e5bff299a/mx-puppet-discord-master.tar.gz
-SOURCE_SUM=3e774c0a7fe02ba92fb2b2e45edb5601e85dfe1a164cba20b60050bfdcb9c77f
+SOURCE_URL=https://gitlab.com/mx-puppet/discord/mx-puppet-discord/-/archive/fd4402268e9a8b942ca632f3337702862e06d24f/mx-puppet-discord-fd4402268e9a8b942ca632f3337702862e06d24f.tar.gz
+SOURCE_SUM=b0edc4dfe9d3670eb8b53dc229a8fd7fb794ada5fe1f3d3ba4c38b54ce5ca176
 SOURCE_SUM_PRG=sha256sum
 SOURCE_FORMAT=tar.gz
 SOURCE_IN_SUBDIR=true

--- a/doc/DESCRIPTION.md
+++ b/doc/DESCRIPTION.md
@@ -1,4 +1,4 @@
-This is a discord puppeting bridge for matrix. It handles bridging private and group DMs, as well as Guilds (servers). It is based on [mx-puppet-bridge](https://github.com/Sorunome/mx-puppet-bridge).
+This is a discord puppeting bridge for matrix. It handles bridging private and group DMs, as well as Guilds (servers). It is based on [mx-puppet-bridge](https://gitlab.com/mx-puppet/mx-puppet-bridge).
 
 ### Features
 
@@ -20,4 +20,4 @@ This is a discord puppeting bridge for matrix. It handles bridging private and g
 
 ### How to
 
-See <https://github.com/matrix-discord/mx-puppet-discord> for documentation.
+See <https://gitlab.com/mx-puppet/discord/mx-puppet-discord> for documentation.

--- a/doc/DESCRIPTION_fr.md
+++ b/doc/DESCRIPTION_fr.md
@@ -1,4 +1,4 @@
-Ceci est une passerelle vers Discord pour Matrix. Il gère la messagerie privée ou en groupe, tout comme la discussion dans une guilde (serveur Discord). Le projet est basé sur [mx-puppet-bridge](https://github.com/Sorunome/mx-puppet-bridge).
+Ceci est une passerelle vers Discord pour Matrix. Il gère la messagerie privée ou en groupe, tout comme la discussion dans une guilde (serveur Discord). Le projet est basé sur [mx-puppet-bridge](https://gitlab.com/mx-puppet/mx-puppet-bridge).
 
 ### Fonctionnalités
 
@@ -20,4 +20,4 @@ Ceci est une passerelle vers Discord pour Matrix. Il gère la messagerie privée
 
 ### Comment faire
 
-Voir la documentation sur <https://github.com/matrix-discord/mx-puppet-discord> (en anglais).
+Voir la documentation sur <https://gitlab.com/mx-puppet/discord/mx-puppet-discord> (en anglais).

--- a/manifest.json
+++ b/manifest.json
@@ -7,7 +7,7 @@
         "fr": "Passerelle Discord pour Matrix/Synapse."
     },
     "version": "0.1.7~ynh1",
-    "url": "https://github.com/matrix-discord/mx-puppet-discord",
+    "url": "https://gitlab.com/mx-puppet/discord/mx-puppet-discord",
     "upstream": {
         "license": " Apache-2.0",
         "website": "https://gitlab.com/mx-puppet/discord/mx-puppet-discord",

--- a/manifest.json
+++ b/manifest.json
@@ -6,13 +6,13 @@
         "en": "A Discord puppeting bridge for Matrix/Synapse.",
         "fr": "Passerelle Discord pour Matrix/Synapse."
     },
-    "version": "0.1.6~ynh2",
+    "version": "0.1.7~ynh1",
     "url": "https://github.com/matrix-discord/mx-puppet-discord",
     "upstream": {
         "license": " Apache-2.0",
-        "website": "https://github.com/matrix-discord/mx-puppet-discord",
-        "admindoc": "https://github.com/matrix-discord/mx-puppet-discord",
-        "code": "https://github.com/matrix-discord/mx-puppet-discord"
+        "website": "https://gitlab.com/mx-puppet/discord/mx-puppet-discord",
+        "admindoc": "https://gitlab.com/mx-puppet/discord/mx-puppet-discord",
+        "code": "https://gitlab.com/mx-puppet/discord/mx-puppet-discord"
     },
     "license": "Apache-2.0",
     "maintainer": {

--- a/scripts/_common.sh
+++ b/scripts/_common.sh
@@ -7,7 +7,7 @@
 # dependencies used by the app
 pkg_dependencies="postgresql"
 # Node version
-NODEJS_VERSION=14
+NODEJS_VERSION=16
 
 #=================================================
 # PERSONAL HELPERS

--- a/scripts/install
+++ b/scripts/install
@@ -130,6 +130,7 @@ popd
 # SETUP PIP (FOR YQ)
 #=================================================
 
+ynh_script_progression --message="Install yq..." --weight=1
 pip_path=$final_path/.pip
 python3 -m venv $pip_path
 $pip_path/bin/pip3 install yq

--- a/scripts/install
+++ b/scripts/install
@@ -123,7 +123,7 @@ ynh_script_progression --message="Building Node dependencies..." --weight=30
 
 pushd "$final_path"
     ynh_use_nodejs
-    ynh_exec_warn_less sudo -u $app env $ynh_node_load_PATH npm install
+    ynh_exec_warn_less sudo -u $app env $ynh_node_load_PATH GIT_SSH_COMMAND='ssh -o StrictHostKeyChecking=accept-new' npm install
 popd
 
 #=================================================

--- a/scripts/upgrade
+++ b/scripts/upgrade
@@ -133,7 +133,7 @@ ynh_script_progression --message="Building Node dependencies..." --weight=3
 
 pushd "$final_path"
     ynh_use_nodejs
-    ynh_exec_warn_less sudo -u $app env $ynh_node_load_PATH npm install
+    ynh_exec_warn_less sudo -u $app env $ynh_node_load_PATH GIT_SSH_COMMAND='ssh -o StrictHostKeyChecking=accept-new' npm install
 popd
 
 #=================================================

